### PR TITLE
dropzone implemented

### PR DESCRIPTION
**Pull request by Martin1509**

### CODE REVIEW

File: src/app/components/App/Form.tsx [hunk 1]
```diff
@@ -144,6 +160,14 @@ const Form: React.FC = () => {
                   onBlur={handleBlur}
                 />
               </div>
+              <Dropzone setFileSelected={setFileSelected} setImagePreview={setImagePreview}/>
+              <div css={inputStyle}>
```
Comment (kingjulien) [inline src/app/components/App/Form.tsx:175]: as you have own Dropzone component, would be nice moving this part into it.
so implementation is easier and not get duplicated

File: src/app/components/App/Form.tsx [hunk 2]
```diff
@@ -144,6 +160,14 @@ const Form: React.FC = () => {
...
+                {imagePreview != undefined && imagePreview != '' ? (
```
Comment (kingjulien) [inline src/app/components/App/Form.tsx:176]: checking against '' should be enough, as you have initial value as empty string.
but please (always!) use type comparison:
`imagePreview !== ''`

File: src/app/components/App/Form.tsx [hunk 3]
```diff
@@ -144,6 +160,14 @@ const Form: React.FC = () => {
...
+                  <img css={{height: '150px'}} src={imagePreview} alt="img"/>
+                ) : (
+                  <></>
```
Comment (kingjulien) [inline src/app/components/App/Form.tsx:179]: no need to render just empty wrapper:
```
{ imagePreview !== '' && (
   <img css={{height: '150px'}} src={imagePreview} alt="img"/>
)}
```

File: src/app/components/App/Dropzone.tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+      reader.onerror = () => console.log('file reading has failed');
+      reader.onload = (readerEvent: ProgressEvent<FileReader>) => {
+
+        //const binaryStr = reader.result;
+        if (file.type.includes('image')) {
+          setImagePreview(readerEvent.target!.result as string);
+        }
+      };
+      reader.readAsArrayBuffer(file);
+      setFileSelected(file);
```
Comment (kingjulien) [inline src/app/components/App/Dropzone.tsx:46]: not sure what exactly this is meant for, but should not be in the loop

File: src/app/components/App/Dropzone.tsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+// });
+
+interface DropzoneProps {
+  setFileSelected: (file: File | undefined) => void;
+  setImagePreview: (file: string) => void;
+}
+
+const Dropzone: React.FC<DropzoneProps> = ({setFileSelected, setImagePreview}) => {
+  const onDrop = useCallback((acceptedFiles) => {
+    acceptedFiles.forEach((file) => {
```
Comment (jozzi) [inline src/app/components/App/Dropzone.tsx:32]: this implementation would only work for the single file even though loops over the multiple files as would call callbacks on every file load.

Better would be either to directly use only first accepted file or change the logic to call callbacks once all of them are loaded with array of files / previews.

Comment (jozzi) [reply src/app/components/App/Dropzone.tsx:32]: as well you can set dropzone explicitly to accept only single file :)

File: src/app/components/App/Form.tsx [hunk 6]
```diff
@@ -58,17 +70,29 @@ const FormSchema = Yup.object().shape({
...
         <Formik
           initialValues={{email: '', name: '', info: '', birthdate: new Date()}}
           validationSchema={FormSchema}
           onSubmit={(values, {setSubmitting}) => {
-            setTimeout(() => {
-              alert(JSON.stringify(values, null, 2));
-              setSubmitting(false);
-            }, 400);
+            axios.post('http://localhost:8081/attendees', {
+              id: 'uuid' + Math.floor(Math.random() * 1000),
```
Comment (jozzi) [inline src/app/components/App/Form.tsx:85]: better is to use some library for uuid as naive implementation will create collisions for bigger datasets

File: src/app/components/App/Form.tsx (hunk 2 above)
Comment (jozzi) [reply src/app/components/App/Form.tsx:176]: yeah that is important especially for numbers as xy != undefined would not work for 0

**Review by jozzi — CHANGES_REQUESTED** [review]

**Pull request merged**